arviz/utils/xarray_utils.py

```diff
@@ -78,7 +83,14 @@ def default_varnames_coords_dims(trace, coords, dims):
         dims = {}
 
     for varname in varnames:
-        dims.setdefault(varname, [])
+        if varname not in dims:
+            vals = trace.get_values(varname, combine=False, squeeze=False)
+            vals = np.array(vals)
+            shape_len = len(vals.shape)
+            if shape_len == 2:
+                dims[varname] = []
+            else:
+                dims[varname] = [f"{varname}_dim_{idx}" for idx in range(1, shape_len-2+1)]
```

Comment:
Ah, I see why things were breaking now.  This definitely makes everything "just work" smoothly, which is nice, but I think I'd rather have this throw an exception instead of silently inferring intent here.  I think the analysis we can do are much more interesting if we know which dimensions are shared between variables.

The test case I am looking at is the non-centered eight schools model.  It now works with no exception, but there is no indication that `theta` and `theta_tilde` both have a coordinate for the school.

Happy to leave it as is if you or @aloctavodia have strong feelings!